src/Microsoft.DotNet.Compiler.Common/Executable.cs

```diff
@@ -254,30 +254,34 @@ public void WriteDeps(LibraryExporter exporter)
         public void GenerateBindingRedirects(LibraryExporter exporter)
         {
             var outputName = _outputPaths.RuntimeFiles.Assembly;
+            var configFile = outputName + ".config";
 
             var existingConfig = new DirectoryInfo(_context.ProjectDirectory)
                 .EnumerateFiles()
                 .FirstOrDefault(f => f.Name.Equals("app.config", StringComparison.OrdinalIgnoreCase));
 
-            XDocument baseAppConfig = null;
-
             if (existingConfig != null)
             {
-                using (var fileStream = File.OpenRead(existingConfig.FullName))
-                {
-                    baseAppConfig = XDocument.Load(fileStream);
-                }
+                File.Copy(existingConfig.FullName, configFile, true);
             }
 
-            var appConfig = exporter.GetAllExports().GenerateBindingRedirects(baseAppConfig);
-
-            if (appConfig == null) { return; }
+            List<string> configFiles = new List<string>();
+            configFiles.Add(configFile);
 
-            var path = outputName + ".config";
-            using (var stream = File.Create(path))
+            foreach (var export in exporter.GetDependencies())
             {
-                appConfig.Save(stream);
+                var dependencyExecutables = export.RuntimeAssemblyGroups.GetDefaultAssets()
+                                                .Where(asset => asset.FileName.ToLower().EndsWith(".exe"))
```

Comment:
We have a constant somewhere with `.exe` value, use it. And change `.config` to use one to.